### PR TITLE
Fix: Comments issues empty statement, empty file and following augment decorator

### LINKED ISSUE #1448: Cannot format empty file with only a comment
```
["ERROR" - 8:21:37 AM] Error formatting document.
["ERROR" - 8:21:37 AM] Cannot read properties of undefined (reading 'trim')
TypeError: Cannot read properties of undefined (reading 'trim')
	at Object.ensureAllCommentsPrinted (/Users/timotheeguerin/dev/azsdk/cadl-azure/common/temp/node_modules/.pnpm/prettier@2.7.1/node_modules/prettier/index.js:8109:55)
````

[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=Ly8gZm9ybWF0)

### LINKED ISSUE #2068: Formatter crash when having trailing comment on augment decorator
```
model Foo {
  foo: string;
}
@@projectedName(Foo.foo, "json", "bar");  // Prefer a different name for java only

```

[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=bW9kZWwgRm9vIHsKICBmb286IHN0cmluZzsKfQpAQHByb2plY3RlZE5hbWUoRm9vLmZvbywgImpzb24iLCAiYmFyIik7ICAvLyBQcmVmZXIgYSBkaWZmZXJlbnQgbmFtZSBmb3IgamF2YSBvbmx5Cg%3D%3D)

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix  #2068
fix #1448

### CODE REVIEW

File: packages/compiler/test/formatter/formatter.test.ts [hunk 1]
```diff
@@ -1687,6 +1723,31 @@ alias Foo = [
     });
   });
 
+  describe("empty statements", () => {
+    it("remove empty statements", () => {
+      const foo = 123; /** abc def ghi */ /** abc def ghi */
```
Comment (mikeharder) [inline packages/compiler/test/formatter/formatter.test.ts:1728]:
```suggestion
```

Should this line just be deleted?  It's causing a lint failure:

```
  1728:13  warning  'foo' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=2848158&view=logs&j=04301bb2-aad8-5bfd-762c-8056a87f3309&t=5f153439-c24b-5508-ce0c-d3e9dc372bb4&l=177

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: packages/compiler/test/formatter/formatter.test.ts [hunk 2]
```diff
@@ -1687,6 +1723,31 @@ alias Foo = [
     });
   });
 
+  describe("empty statements", () => {
+    it("remove empty statements", () => {
+      const _foo = 123; /** abc def ghi */ /** abc def ghi */
```
Comment (mikeharder) [inline packages/compiler/test/formatter/formatter.test.ts:1728]:
```suggestion
```

This line was causing a lint failure:

```
  1728:13  warning  'foo' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=2848158&view=logs&j=04301bb2-aad8-5bfd-762c-8056a87f3309&t=5f153439-c24b-5508-ce0c-d3e9dc372bb4&l=177

I renamed it to `_foo` to unblock the build.  But should this line just be deleted?

Comment (timotheeguerin) [reply packages/compiler/test/formatter/formatter.test.ts:1728]: Oh yeah, forgot to remove was testing typescript formatter behavior

**Comment by mikeharder** [comment]
> Changes in this PR will be published to the following url to try(check status of TypeSpec Pull Request Try It pipeline for publish status): Playground: https://cadlplayground.z22.web.core.windows.net/prs/2069/
> 
> Website: https://cadlwebsite.z1.web.core.windows.net/prs/2069/

@timotheeguerin: Should these links work?  I get 404.

**Comment by timotheeguerin** [comment]
> > Changes in this PR will be published to the following url to try(check status of TypeSpec Pull Request Try It pipeline for publish status): Playground: https://cadlplayground.z22.web.core.windows.net/prs/2069/
> > Website: https://cadlwebsite.z1.web.core.windows.net/prs/2069/
> 
> @timotheeguerin: Should these links work? I get 404.

They do usually buy I think they pipeline for try it is not running for pr against release branch only main. We should change that

**Review by nguerrera — APPROVED** [review]

**Pull request merged**